modules/common/job/job.go

```diff
@@ -170,14 +170,22 @@ func (j *Job) DoJob(
 	}
 
 	// allow updating TTLSecondsAfterFinished even after the job is finished
-	_, err = controllerutil.CreateOrPatch(ctx, h.GetClient(), job, func() error {
-		job.Spec.TTLSecondsAfterFinished = j.job.Spec.TTLSecondsAfterFinished
-		return nil
-	})
-	if err != nil {
+	job, err = GetJobWithName(ctx, h, j.job.Name, j.job.Namespace)
+	if err != nil && !k8s_errors.IsNotFound(err) {
 		return ctrl.Result{}, err
 	}
 
+	if err == nil {
```

Comment:
this condition looks redundant, err will alsways be nil at this point

Reply:
Yes, this is the meat of the patch. We should not try to update the Job with a new TTL if since the Job is deleted.